kik_unofficial/client.py

```diff
@@ -78,7 +85,12 @@ def _connect(self):
         self.kik_connection_thread.start()
 
     def wait_for_messages(self):
-        self.kik_connection_thread.join()
+        for _ in range(5):
+            self.kik_connection_thread.join()
+            log.info("[+] Connection failed, trying again...")
```

Comment:
Let's just write "Connection has disconnected, reconnecting..." or something like that instead to be more accurate (You don't know for sure it failed)